server/src/main/java/org/apache/iotdb/db/query/control/QueryResourceManager.java

```diff
@@ -90,6 +90,19 @@ public long assignQueryId(boolean isDataQuery) {
     return queryId;
   }
 
+  public long assignCompactionQueryId() {
+    long threadNum;
+    try {
+      threadNum = Long.parseLong((Thread.currentThread().getName().split("-"))[3]);
+    } catch (ArrayIndexOutOfBoundsException e) {
+      // for test
+      threadNum = 0;
+    }
```

Comment:
We must not influence the common path code because of test. So it's better to call `setName` to set the thread name in test.

Reply:
resolved